lib/services/role.go

```diff
@@ -2034,6 +2034,37 @@ type kubernetesClusterLabelMatcher struct {
 	clusterLabels map[string]string
 }
 
+// KubernetesResourceMatcher matches a role against a Kubernetes Resource.
+// Kind is must be stricly equal but namespace and name allow wildcards.
+type KubernetesResourceMatcher struct {
+	types.KubernetesResource
+}
+
+// NewLoginMatcher creates a KubernetesResourceMatcher that checks whether the
+// role's KubeResources match the specified condition.
+func NewKubernetesResourceMatcher(resource types.KubernetesResource) *KubernetesResourceMatcher {
+	return &KubernetesResourceMatcher{
+		resource,
```

Comment:
nit: use named fields, they are more resilient to changes:

```suggestion
		KubernetesResource: resource,
```